src/handles/handleDiagnostic.ts

```diff
@@ -87,6 +84,13 @@ async function handleLinter (
       VscUri.parse(textDocument.uri).path,
       rootPatterns,
     )
+
+    if (config.requiredFiles && config.requiredFiles.length) {
```

Comment:
This is just a nice to have to avoid any unnecessary work. This won't work for things that support eg: reading their config from a `package.json` file, but that can be tackled later.